src/modules/fundingManager/token/ERC20Issuance_blacklist_v1.sol

```diff
@@ -64,61 +71,69 @@ contract ERC20Issuance_Blacklist_v1 is
         string memory name_,
         string memory symbol_,
         uint8 decimals_,
-        uint initialSupply_,
+        uint256 initialSupply_,
         address initialAdmin_
     )
         ERC20Issuance_v1(name_, symbol_, decimals_, initialSupply_, initialAdmin_)
-    {}
-
-    //--------------------------------------------------------------------------
-    // Modifiers
-
-    modifier onlyAllowed(address account_) {
-        require(!isBlacklisted(account_), "Address is blacklisted");
-        _;
+        AccessControl()
+    {
+        _grantRole(DEFAULT_ADMIN_ROLE, initialAdmin_);
+        _grantRole(BLACKLIST_MANAGER_ROLE, initialAdmin_);
     }
 
     //--------------------------------------------------------------------------
     // View Functions
 
     /// @inheritdoc IERC20Issuance_Blacklist_v1
-    function isBlacklisted(address account_)
-        public
-        view
-        override
-        returns (bool isBlacklisted_)
-    {
+    function isBlacklisted(address account_) public view returns (bool) {
         return _blacklist[account_];
     }
 
     //--------------------------------------------------------------------------
     // External Functions
 
+    // Modifier to check for blacklist manager role
+    modifier onlyBlacklistManager() {
+        require(hasRole(BLACKLIST_MANAGER_ROLE, msg.sender), "Caller is not a blacklist manager");
```

Comment:
Please add a custom error for this modifier ser

Reply:
done